pkg/exchange/bitget/stream.go

```diff
@@ -0,0 +1,167 @@
+package bitget
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/c9s/bbgo/pkg/exchange/bitget/bitgetapi"
+	"github.com/c9s/bbgo/pkg/types"
+)
+
+//go:generate callbackgen -type Stream
+type Stream struct {
+	types.StandardStream
+
+	booksEventCallbacks []func(o BookEvent)
+}
+
+func NewStream() *Stream {
+	stream := &Stream{
+		StandardStream: types.NewStandardStream(),
+	}
+
+	stream.SetEndpointCreator(stream.createEndpoint)
+	stream.SetParser(stream.parseWebSocketEvent)
+	stream.SetDispatcher(stream.dispatchEvent)
+	stream.OnConnect(stream.handlerConnect)
+
+	stream.OnBooksEvent(stream.handleBooksEvent)
+	return stream
+}
+
+func (s *Stream) syncSubscriptions(opType WsEventType) error {
+	if opType != WsEventUnsubscribe && opType != WsEventSubscribe {
+		return fmt.Errorf("unexpected subscription type: %v", opType)
+	}
+
+	logger := log.WithField("opType", opType)
```

Comment:
opType log tag might not be so helpful?

Reply:
The op type represents a subscription or unsubscription. If an error occurs, we can identify the type that is causing the problem and address it.